src/Symfony/Component/JsonPath/Functions/MatchFunction.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonPath\Functions;
+
+use Symfony\Component\JsonPath\JsonPathUtils;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ *
+ * @internal
+ */
+#[AsJsonPathFunction(name: 'match')]
+final class MatchFunction extends AbstractJsonPathFunction
+{
+    use JsonPathFunctionArgumentTrait;
+
+    public function __invoke(array $args, mixed $context): bool
+    {
+        $valueResults = $args[0] ?? [];
+        $patternResults = $args[1] ?? [];
+
+        $value = \is_array($valueResults) ? ($valueResults[0] ?? null) : $valueResults;
+        $pattern = \is_array($patternResults) ? ($patternResults[0] ?? null) : $patternResults;
+
+        return match (true) {
+            \is_string($value) && \is_string($pattern) => (bool) @preg_match(\sprintf('/^%s$/u', JsonPathUtils::normalizeRegex($pattern)), $value),
+            default => false,
+        };
```

Comment:
please consider changing to this CS everywhere mach is used:
```suggestion
        return \is_string($value) && \is_string($pattern) && @preg_match(\sprintf('/^%s$/Du', JsonPathUtils::normalizeRegex($pattern)), $value);
```